Marlin/src/gcode/config/M200-M205.cpp

```diff
@@ -59,10 +59,11 @@ void GcodeSuite::M201() {
   const int8_t target_extruder = get_target_extruder_from_command();
   if (target_extruder < 0) return;
 
+  static constexpr uint32_t max_accel[] = DEFAULT_MAX_ACCELERATION;
```

Comment:
I'm not sure about adding this implicit upper limit, where `DEFAULT_MAX_ACCELERATION` is no longer just the default setting value, but also determines the upper limit for `M201`. Some users may want to set the default really low (since it's just the _default_ applied on `M502` and a new firmware flash, after all). The best thing to do for `M201` may be to simply allow it to be set to whatever high value it receives and let the consequences of that bad value be visible. I don't think it is sensible to conceal bad slicer behaviors like this one in the firmware. My idea about "locking" certain settings that slicers shouldn't mess with was kind of tongue-in-cheek, as I really think it's up to slicers to make it more obvious what they are doing, make it easier to turn this behavior off, and turn off these overrides by default.